src/main/java/org/opentripplanner/routing/algorithm/strategies/SimpleEuclideanRWH.java

```diff
@@ -0,0 +1,103 @@
+        calculateDestination(options.rctx.target);
+        calculateBestMultiplier(options);
+    }
+
+    private void calculateDestination(Vertex target) {
+        if (target.getDegreeIn() == 1) {
+            Optional<Edge> edge = target.getIncoming().stream().findFirst().filter(FreeEdge.class::isInstance);
+            if (edge.isPresent()) {
+                lat = edge.get().getFromVertex().getLat();
+                lon = edge.get().getFromVertex().getLon();
+                return;
+            }
+        }
+        lat = target.getLat();
+        lon = target.getLon();
+    }
+
+    private void calculateBestMultiplier(RoutingRequest options) {
+        bestMultiplier = options.routingReluctances.getWalkReluctance() / options.walkSpeed;
+        // max speed of transit in not higher than max speed of a car, and transit does not have a reluctance
+        if (options.modes.isTransit()) {
+            bestMultiplier = min(bestMultiplier, 1.0 / options.carSpeed);
+        }
+        if (options.rentingAllowed) {
+            calculateBestMultiplierWithRentableVehicles(options);
+        } else {
+            calculateBestMultiplierWithOwnVehicles(options);
+        }
+    }
+
+    private void calculateBestMultiplierWithRentableVehicles(RoutingRequest options) {
+        Set<VehicleType> vehicleTypesAllowed = options.vehicleValidator.getVehicleTypesAllowed();
+        if (vehicleTypesAllowed.contains(VehicleType.CAR)) {
+            bestMultiplier = min(bestMultiplier, options.routingReluctances.getCarReluctance() / CarDescription.getMaxPossibleSpeed());
+        }
+        if (vehicleTypesAllowed.contains(VehicleType.MOTORBIKE)) {
+            bestMultiplier = min(bestMultiplier, options.routingReluctances.getMotorbikeReluctance() / MotorbikeDescription.getMaxPossibleSpeed());
+        }
+        if (vehicleTypesAllowed.contains(VehicleType.KICKSCOOTER)) {
+            bestMultiplier = min(bestMultiplier, options.routingReluctances.getKickScooterReluctance() / KickScooterDescription.getMaxPossibleSpeed());
+        }
+        if (vehicleTypesAllowed.contains(VehicleType.BIKE)) {
+            bestMultiplier = min(bestMultiplier, options.routingReluctances.getBicycleReluctance() / BikeDescription.getMaxPossibleSpeed());
+        }
+    }
+
+    private void calculateBestMultiplierWithOwnVehicles(RoutingRequest options) {
+        if (options.modes.getCar()) {
+            bestMultiplier = min(bestMultiplier, options.routingReluctances.getCarReluctance() / options.carSpeed);
+        }
+        if (options.modes.getBicycle()) {
+            bestMultiplier = min(bestMultiplier, options.routingReluctances.getBicycleReluctance() / options.bikeSpeed);
+        }
+    }
+
+    /**
+     * Estimate remaining weight by multiplying euclidean distance to destination
+     * by lowest multiplier for all possible traverse modes.
+     * Multiplier is just reluctance divided by speed.
```

Comment:
może przenieś ten komentarz na całą klasę?
dobry tip do inelij to ctrl+J by wyświetlić dokumentację.
Heurystyka wydaje się tego typu rzeczą, że chyba warto udokumentować ideę w niej zawartą.

Reply:
done